src/benchttp/run.ts

```diff
@@ -0,0 +1,55 @@
+import { RunConfiguration } from './configuration'
+import { MetricField } from './metrics'
+import { TestPredicate } from './tests'
+
+export interface RunError {
+  error: string
+}
+
+export interface RunProgress {
+  done: boolean
+  doneCount: number
+  maxCount: number
+  timeout: number
+  elapsed: number
+}
+
+export interface RunReport {
+  metrics: {
+    responseTimes: TimeStats
+    requestEventTimes: TimeStats
+    statusCodesDistribution: Record<string, number>
+    records: { responseTime: number }[]
+    requestFailures: { reason: string }[]
+  }
+  tests: {
+    pass: boolean
+    results: {
+      input: RunTestCase
+      pass: boolean
+      summary: string
+    }[]
+  }
+  metadata: {
+    config: RunConfiguration
+    startedAt: number
+    finishedAt: number
+  }
+}
+
+interface TimeStats {
```

Comment:
Suggestion: `Stats` could be enough as there is no reference to time data here, simply numbers.